Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -322,6 +322,23 @@ theorem eq_prod_roots_of_splits_id {p : K[X]} (hsplit : Splits (RingHom.id K) p)
     p = C p.leadingCoeff * (p.roots.map fun a => X - C a).prod := by
   simpa using eq_prod_roots_of_splits hsplit
 
+theorem Splits.dvd_of_roots_le_roots {p q : K[X]} (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0)
+    (hq : p.roots ≤ q.roots) : p ∣ q := by
+  by_cases hq0 : q = 0
+  · simp [hq0]
+  · rw [eq_prod_roots_of_splits_id hp, C_mul_dvd (leadingCoeff_ne_zero.2 hp0)]
+    exact dvd_trans
+      (Multiset.prod_dvd_prod_of_le (Multiset.map_le_map hq))
+      (prod_multiset_X_sub_C_dvd _)
+
+theorem Splits.dvd_iff_roots_le_roots {p q : K[X]}
+    (hp : p.Splits (RingHom.id _)) (hp0 : p ≠ 0) (hq0 : q ≠ 0) :
+    p ∣ q ↔ p.roots ≤ q.roots := by
+  refine ⟨?_, hp.dvd_of_roots_le_roots hp0⟩
```

Comment:
Can you reuse [Polynomial.roots.le_of_dvd](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Roots.html#Polynomial.roots.le_of_dvd) here? Also, since each implication has weaker assumption, I would mention both of them in the docstring.

Reply:
Yes, this is now fixed